### ISSUE TITLE
Too few days' data kept in March if preserving 31 days worth of data

### ISSUE THREAD

**Issue by morgant** [morgant/43f#2]
There's a logic bug in detecting wrapping ranges in days to keep in `roll_repository_directories()` which affects March if preserving more than 28/29 days worth of data (depending on whether it is a leap year).

This is because `keep_through` will be less than today's date (because 28th/29th through the nonexistent 31st in day in February will be skipped), failing checks for wrapping ranges. Currently, wrapping range detection expects `keep_through` to be higher than today's date, which won't be the case when keeping more days than the number of days in the previous month (e.g. March following February).

This results in only a couple days' worth of data being preserved, which is the opposite of what is expected.

**Comment by morgant** [comment]
I _think_ we could resolve this in `roll_repository_directories()` by setting `keep_through` to tomorrow's date if `config_keep_days` is greater than the number of days in the previous month. I need to verify this.

We should be able to determine the number of days in the previous month, by selecting the first day of the current month and using adjusting -1 days. This might require implementing changes to `reldate()` or an entirely new method.

**Comment by morgant** [comment]
I believe my previous note about this occurring when `config_keep_days` is greater than the number of days in the previous month was correct. I've implemented a fix in the `too-few-days-data-kept-in-march` branch, which could use more testing. Specifically, if `config_keep_days` is greater than the number of days in the previous month, then it just uses the number of days in the previous month in the calculation of the `keep_through` day. I believe this to be safe.

To get to this fix, I've replaced the `reldate()` function with a new `relative_to_now()` function, which--in turn--is powered by a new `relative_date()` function. There's also a new `days_in_previous_month()` function which also uses `relative_date()`. All of this relies solely on Dateutils (it' wasn't feasible without it), so I've noted that it is a requirement for all platforms.
